gadgets/trace_dns/program.bpf.c

```diff
@@ -300,7 +300,16 @@ int ig_trace_dns(struct __sk_buff *skb)
 		if (skb->len <= dns_off)
 			return 0;
 
-		// DNS is after the TCP header and the 2 bytes of the length of the DNS packet
+		// DNS messages over TCP are prefixed with a two byte length field which
```

Comment:
Mmmm I analyzed it more and I think we can do something better. Given that TCP is a stream-based protocol, if there is a big DNS query/response, the fragmentation will not happen at the TCP level, but at the IP level. So, I was wondering if we can check the _Fragment Offset Field_ in the IP header to identify whether the packet is a fragment, i.e., if this field is non-zero, it indicates that the packet is a fragment. WDYT?